webapp/src/components/App/Studies/StudyTree/index.tsx

```diff
@@ -0,0 +1,298 @@
+  workspace: string,
+) {
+  const emptyNode = { name: workspace, path: `/${workspace}`, children: [] };
+  if (stydyTree.children.some((child) => child.name === workspace)) {
+    return stydyTree;
+  }
+  return {
+    ...stydyTree,
+    children: [...stydyTree.children, emptyNode],
+  };
+}
+
+/**
+ * Insert several workspaces into the study tree if they don't exist already in the tree.
+ *
+ * @param workspaces workspaces to insert into the tree
+ * @param stydyTree study tree to insert the workspaces into
+ * @returns study tree with the empty workspaces inserted if they weren't already there.
+ */
+export function insertWorkspacesIfNotExist(
+  stydyTree: StudyTreeNode,
+  workspaces: string[],
+): StudyTreeNode {
+  return workspaces.reduce((acc, workspace) => {
+    return insertWorkspaceIfNotExist(acc, workspace);
+  }, stydyTree);
+}
+
+/**
+ * fetch and insert the workspaces into the study tree.
+ * @param studyTree study tree to insert the workspaces into
+ * @returns study tree with the workspaces inserted if they weren't already there.
+ */
+async function fetchAndInsertWorkspaces(
+  studyTree: StudyTreeNode,
+): Promise<StudyTreeNode> {
+  const workspaces = await api.getWorkspaces();
+  return insertWorkspacesIfNotExist(studyTree, workspaces);
+}
+
+function StudyTree() {
+  const initialStudiesTree = useAppSelector(getStudiesTree);
+  const [studiesTree, setStudiesTree] = useState(initialStudiesTree);
+  const folder = useAppSelector((state) => getStudyFilters(state).folder, R.T);
+  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
+  const dispatch = useAppDispatch();
+
+  // Initialize folders once we have the tree
+  useUpdateEffectOnce(() => {
+    fetchAndInsertWorkspaces(initialStudiesTree)
+      .then((tree) => {
+        // once we get the workspaces we intialize the first level of folders
+        const chidrenPaths = tree.children.map((child) => `root${child.path}`);
+        fetchAndInsertSubfolders(chidrenPaths, tree).then((r) => {
+          setStudiesTree(r[0]);
+          for (const path of r[1]) {
+            enqueueErrorSnackbar(
+              `Failed to initialize folders for : ${path}`,
+              "details in console.error",
+            );
+          }
+        });
+      })
+      .catch((error) => {
+        enqueueErrorSnackbar("Failed to load list workspaces", error);
+        setStudiesTree(initialStudiesTree);
+      });
+  }, [initialStudiesTree]);
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleTreeItemClick = (
+    itemId: string,
+    studyTreeNode: StudyTreeNode,
+  ) => {
+    dispatch(updateStudyFilters({ folder: itemId }));
+    if (itemId === "root") {
+      fetchAndInsertWorkspaces(studiesTree)
+        .then(setStudiesTree)
+        .catch((error) => {
+          enqueueErrorSnackbar("Failed to load list workspaces", error);
```

Comment:
Add i18n key

Reply:
i18n key still missing, in addition please consider refactoring this event handler to use async/await syntax